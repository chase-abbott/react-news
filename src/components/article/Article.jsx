import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ article }) => (
  <div style={{ marginLeft: '5px' }}>
    <p> {article.title}</p>
    <p> {article.author}</p>
    <span> {article.description}</span>
  </div>
);

Article.propTypes = {
  article: PropTypes.object.isRequired
};

export default Article;
