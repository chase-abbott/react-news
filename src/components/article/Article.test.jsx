import React from 'react';
import { render } from '@testing-library/react';
import Article from './Article.jsx';

describe('ArticleList componenet', () => {
  it('renders a list of articles', () => {
    const article = {
      title: 'Sports team wins',
      author: 'Chase Abbott',
      description: 'They won by a lot'
    };

    const { asFragment } = render(<Article article={article}/>);

    expect(asFragment()).toMatchSnapshot();
  });
});
