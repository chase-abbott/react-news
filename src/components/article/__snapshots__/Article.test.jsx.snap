// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article componenet renders a title, author, and description 1`] = `
<DocumentFragment>
  <div>
    <p>
       Sports team wins
    </p>
    <p>
       Chase Abbott
    </p>
    <span>
       They won by a lot
    </span>
  </div>
</DocumentFragment>
`;
